kernelci/cli/__init__.py

```diff
@@ -0,0 +1,42 @@
+# Copyright (C) 2022 Collabora Limited
+# Author: Guillaume Tucker <guillaume.tucker@collabora.com>
+#
+# This module is free software; you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the Free
+# Software Foundation; either version 2.1 of the License, or (at your option)
+# any later version.
+#
+# This library is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this library; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+import sys
+
+from .base import Args, Command, parse_opts, sub_main
+from . import validate
+
+_COMMANDS = {
+    'validate': validate.main,
+}
+
+
+def list_command_names():
+    """Return a list with the command names"""
+    return _COMMANDS.keys()
```

Comment:
Either docstring or implementation is wrong. `.keys()` return type is not `list`.

```python
>>> d = {'a': 1}
>>> isinstance(d.keys(), list)
False
>>> type(d.keys())
<class 'dict_keys'>


```

Reply:
```
keys(...)
    D.keys() -> a set-like object providing a view on D's keys
```
Well it's not strictly a Python list but it's close enough and still kind of true in terms of semantics.  I could just say, "Return the command names".  Or `list()` it.  But what's your opinion?

Reply:
OK I've turned the returned value into an actual `list` object.

Reply:
Thanks for this change, I guess it's fine to return an actual list here. 
I don't have strong opinions on what kind of object it should be, I just wanted to have the description match the actual object. The list in python is both iterable and subscriptable, while set-like object is only subscriptable and is not meant to keep it's elements order.